presto-main/src/main/java/com/facebook/presto/operator/aggregation/IntervalDayToSecondSumAggregation.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation;
+
+import com.facebook.presto.operator.aggregation.state.NullableLongState;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.function.AggregationFunction;
+import com.facebook.presto.spi.function.CombineFunction;
+import com.facebook.presto.spi.function.InputFunction;
+import com.facebook.presto.spi.function.OutputFunction;
+import com.facebook.presto.spi.function.SqlType;
+import com.facebook.presto.type.BigintOperators;
+
+import static com.facebook.presto.spi.type.StandardTypes.INTERVAL_DAY_TO_SECOND;
+import static com.facebook.presto.type.IntervalDayTimeType.INTERVAL_DAY_TIME;
+
+@AggregationFunction("sum")
+public final class IntervalDayToSecondSumAggregation
+{
+    private IntervalDayToSecondSumAggregation() {}
+
+    @InputFunction
+    public static void sum(NullableLongState state, @SqlType(INTERVAL_DAY_TO_SECOND) long value)
+    {
+        state.setNull(false);
+        state.setLong(BigintOperators.add(state.getLong(), value));
+    }
+
+    @CombineFunction
+    public static void combine(NullableLongState state, NullableLongState otherState)
+    {
+        if (state.isNull()) {
+            state.setNull(false);
+            state.setLong(otherState.getLong());
+            return;
+        }
+
+        state.setLong(BigintOperators.add(state.getLong(), otherState.getLong()));
```

Comment:
I think otherState cannot be null, it's guaranteed by `AccumulatorCompiler.generateBlockNonNullPositionForLoop` and required by `AccumulatorStateSerializer.deserialize` and it's usage in `AccumulatorCompiler.generateAddIntermediateAsCombine`.

Also if it wasn't guaranteed there are many places where this is done wrong, if we would like to resign from this assumption I would suggest to fix that in all places at the same time.